Packs/Base/Scripts/CommonServerPython/CommonServerPython.py

```diff
@@ -12270,8 +12283,127 @@ def get_server_config():
     return server_config
 
 
-from DemistoClassApiModule import *     # type:ignore [no-redef]  # noqa:E402
+def xsoar_profiler(func):
+    """
+    A decorator that profiles the execution of a function and generates a profiling report.
+    To use, decorate the function that calls the function you want to profile with @xsoar_profiler.
+    Example: I want to profile the function_to_profile() function:
+        function_to_profile():
+            # some code
+
+        @xsoar_profiler
+        foo():
+            function_to_profile()
+
+    :param func: The function to be profiled.
+    :return: The profiled function.
+    """
+    import cProfile
+    import threading
+
+    def profiler_wrapper(*args, **kwargs):
+        """
+        A wrapper function that profiles the execution of the decorated function.
+        :param args: The positional arguments to be passed to the decorated function.
+        :param kwargs: The keyword arguments to be passed to the decorated function.
+        :return: The result of the decorated function.
+        """
+
+        def profiler_function(signal_event, profiler):
+            """
+            A helper function that runs the profiled function. When the profiled function is finished
+             a signal is received and the profiling data is dumped to a temporary file.
+             Otherwise, the function will stop when reaching to timeout.
+            :param signal_event: A signal that will be set when the profiled function is finished.
+            :param profiler: The Profile object to use for profiling.
+            """
+
+            def dump_result():
+                """
+                Helper function to dump the profiling results to a file and return the file.
+                """
+                import tempfile
+                # Create a temporary file to store profiling data
+                with tempfile.NamedTemporaryFile(delete=False) as temp_file:
+                    profiler.dump_stats(temp_file.name)
+                    temp_file_path = temp_file.name
+
+                # Read the profiling data from the temporary file
+                with open(temp_file_path, 'rb') as f:
+                    profiling_results = f.read()
```

Comment:
double check we cant write to a StringIO or something as a file instead, so we dont need the overhead of writing to disk

Reply:
The StringIO is creating the desired file but seems to be corrupted as I can't open it with the `snakeviz`.